lib/resty/acme/storage/redis.lua

```diff
@@ -42,9 +42,14 @@ local function op(self, op, ...)
   if not ok then
     return nil, err
   end
-  
+
   if self.auth then
-    local _, err = client:auth(self.auth)
+    local _, err
+    if type(self.auth) == "table" then
+      _, err = client:auth(self.auth.username, self.auth.password)
```

Comment:
I would actually suggest new fields being added (username, password) in addition to `auth`, otherwise we may need to write complex compat code on kong side.

Reply:
I think from style perspective I'd rather prefer to have separate fields as well. However I'm wondering - will it really require compat code on kong side? Even the way it's implemented here? 🤔 CP/DP will run different versions of this lib so I think we're good, right?

Reply:
we do, unfortunately, because we need to define the schema at kong side (https://github.com/Kong/kong/blob/master/kong/plugins/acme/schema.lua#L85).
actually, i'm thinking to implement a new redis storage, using the kong provided redis library + schema in the kong acme plugin, that skipped the redis storage provided by this library.
basically we only need to add a `kong/plugins/acme/storage/redis.lua` then do something similar as this line https://github.com/Kong/kong/blob/master/kong/plugins/acme/client.lua#L91

Reply:
done here https://github.com/fffonion/lua-resty-acme/pull/121/commits/376643327496f81ee50579538c30490c1f0a4551

Reply:
@fffonion Please correct me if I'm wrong but I think with the introduction of config adapter: https://github.com/Kong/kong/blob/master/kong/plugins/acme/storage/config_adapters/redis.lua
There's no need to write any compat code since plugin's schema is separated from this library's config.